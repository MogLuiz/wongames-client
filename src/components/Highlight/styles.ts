// Packages
import styled, { css } from "styled-components"

// Types
import { HighlightProps } from "."

type WrapperProps = Pick<HighlightProps, "backgroundImage">

export const Wrapper = styled.section<WrapperProps>`
  ${({ theme, backgroundImage }) => css`
    position: relative;
    height: 23rem;
    display: grid;

    background-image: url(${backgroundImage});
    background-position: center center;
    background-size: cover;

    &::after {
      content: "";
      position: absolute;
      width: 100%;
      height: 100%;
      background-color: rgba(0, 0, 0, 0.6);
    }
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    z-index: ${theme.layers.base};
    text-align: right;
    padding: ${theme.spacings.xsmall};
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.bold};
    color: ${theme.colors.white};
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.light};
    color: ${theme.colors.white};
    margin-bottom: ${theme.spacings.medium};
  `}
`
