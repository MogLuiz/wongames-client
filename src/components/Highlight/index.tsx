// Packages
import React from "react"

// Components
import Button from "components/Button"

// Styles
import * as S from "./styles"

export type HighlightProps = {
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  backgroundImage: string
}

const Highlight: React.FC<HighlightProps> = ({
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  backgroundImage
}) => (
  // -------------------------------------------------
  // Render
  // -------------------------------------------------
  <S.Wrapper backgroundImage={backgroundImage}>
    <S.Content>
      <S.Title>{title}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
      <Button as="a" href={buttonLink}>
        {buttonLabel}
      </Button>
    </S.Content>
  </S.Wrapper>
)

export default Highlight
