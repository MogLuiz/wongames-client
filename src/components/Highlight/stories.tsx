// Packages
import { Story, Meta } from "@storybook/react/types-6-0"

// Components
import Highlight, { HighlightProps } from "."

export default {
  title: "Highlight",
  component: Highlight,
  args: {
    title: "Read Dead est√° de de volta!",
    subtitle: "Venha conhecer as novas aventuras de John",
    buttonLabel: "Buy now",
    buttonLink: "/rdr2"
  }
} as Meta

export const Default: Story<HighlightProps> = (args) => <Highlight {...args} />
