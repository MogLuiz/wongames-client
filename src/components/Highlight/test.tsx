// Packages
import { render, screen } from "@testing-library/react"

// Utils
import { renderWithTheme } from "utils/tests/helpers"

// Components
import Highlight from "."

const props = {
  title: "Heading 1",
  subtitle: "Heading 2"
}

describe("<Highlight />", () => {
  it("should render heading and button", () => {
    const { container } = renderWithTheme(<Highlight {...props} />)

    expect(
      screen.getByRole("heading", { name: /heading 1/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole("heading", { name: /heading 2/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
