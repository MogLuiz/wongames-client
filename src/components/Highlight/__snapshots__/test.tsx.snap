// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Highlight /> should render heading and button 1`] = `
<main
  class=""
>
  <h1>
    Highlight
  </h1>
</main>
`;

exports[`<Highlight /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Highlight
  </h1>
</main>
`;
